core/src/main/java/org/apache/brooklyn/core/mgmt/persist/FileBasedObjectStore.java

```diff
@@ -94,25 +94,19 @@ public File getBaseDir() {
     }
     
     public void prepareForMasterUse() {
-        if (doneFirstContentiousWrite.get())
-            return;
-        synchronized (this) {
-            if (doneFirstContentiousWrite.get())
-                return;
-            try {
-                if (deferredBackupNeeded) {
-                    // defer backup and path creation until first write
-                    // this way if node is standby or auto, the backup is not created superfluously
+        if (doneFirstContentiousWrite.getAndSet(true)) return;
```

Comment:
@aledsage you're right that the semantics changed here. in fact, I think all we need is a normal boolean and a `synchronized` critical section, the atomic CAS aspect of the `AtomicBoolean` aren't being used here, since we can be sure only one thread at a time is checking the value inside the critical section. I've refactored to reflect that.